spyder/api/plugins.py

```diff
@@ -304,115 +288,506 @@ def register_shortcut(self, qaction_or_qshortcut, context, name,
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        if self.shortcut is not None and self.main is not None:
+            sc = QShortcut(QKeySequence(self.shortcut), self.main,
+                           self.switch_to_plugin)
+            self.register_shortcut(sc, "_", "Switch to {}".format(
+                self.CONF_SECTION))
+
+        return (dock, self._LOCATION)
+
+    def _update_plugin_title(self):
+        """Update plugin title, i.e. dockwidget or window title"""
+        if self.dockwidget is not None:
+            win = self.dockwidget
+        elif self._undocked_window is not None:
+            win = self._undocked_window
+        else:
+            return
+
+        win.setWindowTitle(self.get_plugin_title())
+
+    @Slot()
+    def _plugin_closed(self):
+        """DockWidget was closed."""
+        if self._toggle_view_action:
+            self._toggle_view_action.setChecked(False)
+
+    def _create_toggle_view_action(self):
+        """Associate a toggle view action with each plugin"""
+        title = self.get_plugin_title()
+        if self.CONF_SECTION == 'editor':
+            title = _('Editor')
+
+        if self.shortcut is not None:
+            action = create_action(
+                self,
+                title,
+                toggled=lambda checked: self.toggle_view(checked),
+                shortcut=QKeySequence(self.shortcut),
+                context=Qt.WidgetShortcut,
+            )
+        else:
+            action = create_action(
+                self,
+                title,
+                toggled=lambda checked: self.toggle_view(checked),
+            )
+
+        self._toggle_view_action = action
+
+    @Slot()
+    def _close_window(self):
+        """Close QMainWindow instance that contains this plugin."""
+        if self._undocked_window is not None:
+            self._undocked_window.close()
+            self._undocked_window = None
+
+            # Oddly, these actions can appear disabled after the Dock
+            # action is pressed
+            self._undock_action.setDisabled(False)
+            self._close_plugin_action.setDisabled(False)
+
+    @Slot()
+    def _create_window(self):
+        """Create a QMainWindow instance containing this plugin."""
+        self._undocked_window = window = PluginWindow(self)
+        window.setAttribute(Qt.WA_DeleteOnClose)
+        icon = self.get_plugin_icon()
+
+        if is_text_string(icon):
+            icon = self.get_icon(icon)
+
+        window.setWindowIcon(icon)
+        window.setWindowTitle(self.get_plugin_title())
+        window.setCentralWidget(self)
+        window.resize(self.size())
+        self.refresh_plugin()
+        self.set_ancestor(window)
+        self.dockwidget.setFloating(False)
+        self.dockwidget.setVisible(False)
+
+        window.show()
+
+    # FIXME: Now this method sounds like a signal, not a method.
```

Comment:
I think `on_xxx` sounds like a handler for `xxx`, but I guess `handle_xxx` is more explicit.

Reply:
Fixed